src/NUnitFramework/tests/Internal/TestExecutionContextTests.cs

```diff
@@ -984,6 +984,16 @@ private class TestClass
         }
 #endif
 
+        #endregion
+
+#region CurrentRepeatCount Tests
+        [Test]
+        public void CanAccessCurrentRepeatCount()
+        {
+            Assert.That(_fixtureContext.CurrentRepeatCount, Is.EqualTo(0), "expected value to default to zero");
+            _fixtureContext.CurrentRepeatCount++;
+            Assert.That(_fixtureContext.CurrentRepeatCount, Is.EqualTo(1), "expected value to be able to be incremented from the TestExecutionContext");
```

Comment:
Second assert is a test of C#!

Reply:
I put it there just to ensure that the property remains able to be set (i.e. so someone doesn't change this in the future without causing a failure). Will happily change it to some other mechanism to verify that if you'd like

Reply:
Not sure what you are suggesting might happen... if the property were not settable, then NUnit would not compile. It's just an automatic property, right? The test does no harm, however.

Reply:
@CharliePoole, you are correct, in the current implementation (the code I've added to _RetryAttribute_) it would not compile if the setter were removed and in thinking of scenarios where it could be made to still compile, I've convinced myself that it is *VERY* unlikely that someone would go so far as to make such a change unintentionally, so, you're right, that this isn't really needed. Happy to remove it if you would like. As you said though, it does no harm, so either way is ok, I suppose.

Reply:
This test does more than ensure that you can access the count. it also ensures that custom attributes written from scratch (which we have seen) are also able to mutate that count.

Reply:
I have no problem keeping the test, but it doesn't show that at all, since no custom attribute is involved. For example, change the setter to `internal` and the test will still pass, but a custom attribute would be unable to use it.

You bring up a good point though. Maybe we need a set of custom attribute tests using a phony attribute that accesses everything we expose for them to use. Separate issue, however.

Reply:
Oh! I forgot about InternalsVisibleTo. Quite right and an interesting topic.